### PR TITLE
feat: export database data

### PR DESCRIPTION
**Pull request by fengjiachun**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

CLI: support exporting database data with a time range

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [x] This PR requires documentation updates.

## Summary by CodeRabbit

- **New Features**
  - Introduced support for exporting database data, with a new `DatabaseData` option.
  - Deprecated the `TableData` export target.
  
- **Enhancements**
  - Improved the export functionality to handle database data asynchronously for better performance.
  - Added time range specifications for exports, allowing users to set start and end times for greater control over the exported data.

### CODE REVIEW

**Comment by fengjiachun** [comment]
@WenyXu @waynexia PTAL

**Comment by killme2008** [comment]
@fengjiachun  Good job. The task #4276 should be started!

File: src/cmd/src/cli/export.rs [hunk 1]
```diff
@@ -409,14 +426,102 @@ impl Export {
...
+        let db_count = db_names.len();
+        let mut tasks = Vec::with_capacity(db_names.len());
+        for (catalog, schema) in db_names {
+            let semaphore_moved = semaphore.clone();
+            tasks.push(async move {
+                let _permit = semaphore_moved.acquire().await.unwrap();
+                tokio::fs::create_dir_all(&self.output_dir)
+                    .await
+                    .context(FileIoSnafu)?;
+                let output_dir = Path::new(&self.output_dir).join(format!("{catalog}-{schema}/"));
```
Comment (killme2008) [inline src/cmd/src/cli/export.rs:443]: `{catalog}-{schema}` maybe not a good choice. What about organizing the exported table like this:

```
catalog/
     table1/
     table2/
     table2/
```
It's more structural, and avoids the special characters in catalog or table names such as `-`.

Comment (fengjiachun) [reply src/cmd/src/cli/export.rs:443]: Do you mean this?

```
catalog/
     database1/
     database2/
     database2/
```

The current is:

```
     catalog-database1/
     catalog-database2/
     catalog-database2/
```

The smallest export unit is the database, not the table.

Comment (fengjiachun) [reply src/cmd/src/cli/export.rs:443]: We don't support exporting multiple `catalogs` at once, and `catalog` and `schema` names can't contain `-`, so an additional directory level isn't necessary. I think it's fine as is.

Comment (killme2008) [reply src/cmd/src/cli/export.rs:443]: But if the catalog name or database name contains some special characters, there will be some issues.Yes, i meant:

```
catalog/
     database1/
     database2/
     database2/
```

Comment (fengjiachun) [reply src/cmd/src/cli/export.rs:443]: Ok, I'll make the change.

Comment (fengjiachun) [reply src/cmd/src/cli/export.rs]: image

**Review by waynexia — APPROVED** [review]

File: src/cmd/src/cli/export.rs [hunk 2]
```diff
@@ -293,11 +293,14 @@ impl Export {
...
-                tokio::fs::create_dir_all(&self.output_dir)
+                let output_dir = Path::new(&self.output_dir)
+                    .join(&catalog)
+                    .join(format!("{schema}/"));
+                tokio::fs::create_dir_all(&output_dir)
                     .await
                     .context(FileIoSnafu)?;
                 let output_file =
-                    Path::new(&self.output_dir).join(format!("{catalog}-{schema}.sql"));
+                    Path::new(&output_dir).join(format!("{schema}_create_tables.sql"));
```
Comment (killme2008) [inline src/cmd/src/cli/export.rs:303]: Why not just put it into the `schema/` folder?

Comment (killme2008) [reply src/cmd/src/cli/export.rs:303]: Like `{schema}/create_tables.sql`

Comment (fengjiachun) [reply src/cmd/src/cli/export.rs:303]: Good point!

Comment (fengjiachun) [reply src/cmd/src/cli/export.rs]: It’s already in {schema}/; I just removed the file name prefix.

**Pull request merged**